### DISCUSSION TITLE
Fixed code scanning alerts still show up as open

### DISCUSSION THREAD

**Discussion by edumserrano** [community/community#23403] [Code Security]
<p>Hi,</p>
<p>I’ve been reading through the code scanning docs and after configuring it for my repo it flagged some issues which was great. Then I did some pushes to fix some of the issues and they got marked as “Fixed”. However they still show on the “is:open” filter and they count towards the total number of security issues on the “Security tab” of the repo.</p>
<p>On the <a href="https://docs.github.com/en/enterprise-server@2.22/github/finding-security-vulnerabilities-and-errors-in-your-code/automatically-scanning-your-code-for-vulnerabilities-and-errors/managing-code-scanning-alerts-for-your-repository#fixing-an-alert" rel="noopener nofollow ugc">Fixing an alert doc</a> it says:</p>
<p>“If you have write permission for a repository, you can view fixed alerts by viewing the summary of alerts and clicking <strong>Closed</strong>. […] The “Closed” list shows fixed alerts and alerts that users have closed.”</p>
<p>This is happening to me on this <a href="https://github.com/edumserrano/dot-net-sdk-extensions" rel="noopener nofollow ugc">public repo</a> of mine. The filter <code>is:open "Useless assignment to local variable"</code> on code scanning shows what I mean, it returns 3 fixed alerts.</p>
<p>Am I missing something obvious? Shouldn’t the fixed alerts be in the closed state?</p>

**Comment by edumserrano** [comment, marked as answer]
<p>After talking with GitHub support this was the solution I was given that worked for me:</p>
<blockquote>
<p>Hello Eduardo,<br><br>
I heard back from engineering and the reason that the alerts are open is because they are open in a previous stale configuration that is no longer updated. This happened when the analysis file was renamed and left the analyses under the old name stale.<br><br>
This can be detected through the API, by using the list analyses request (<a href="https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-analyses-for-a-repository" class="inline-onebox" rel="noopener nofollow ugc">Code scanning - GitHub Docs</a>). From that response you can see multiple analyses that has “deletable”: true and with different values in the category field.<br><br>
To fix it you should delete all the stale configurations using the delete API (<a href="https://docs.github.com/en/rest/reference/code-scanning#delete-a-code-scanning-analysis-from-a-repository" class="inline-onebox" rel="noopener nofollow ugc">Code scanning - GitHub Docs</a>). Note that if a particular configuration has had several deliveries, then older analyses become deletable when the most recent one is deleted (the delete works as an undo operation) and they also need to be deleted."</p>
</blockquote>
<p>This problem happened to me because I had ran an initial CodeQL scan and then renamed the workflow for CodeQL.</p>
<p>Hope this helps others.</p>

**Comment by ajay-qburst** [comment]
Had the same problem, but the alerts already generated during the initial CodeQL scan was not removed as part of deleting the stale workflow, even though the fix was pushed to the master (or respective) branches. Needed to remove the stale code scanning configurations from those  outdated alerts to get that fixed. Refer: https://github.blog/changelog/2023-03-10-delete-stale-code-scanning-configurations-to-close-outdated-alerts/
